ranger2/aws/ranger_asg/main.tf

```diff
@@ -0,0 +1,579 @@
+  instance_class            = var.db_instance_type
+  allocated_storage         = var.db_allocated_storage
+  name                      = var.db_name
+  username                  = var.db_user
+  password                  = var.db_pwd
+  port                      = var.rds_port
+  db_subnet_group_name      = var.db_subnet_group_name
+  vpc_security_group_ids    = [aws_security_group.rds_sg.id]
+  skip_final_snapshot       = true
+  final_snapshot_identifier = "Ignore"
+  publicly_accessible       = true
+}
+provider "mysql" {
+  endpoint = aws_db_instance.ranger_mysql.endpoint
+  username = aws_db_instance.ranger_mysql.username
+  password = aws_db_instance.ranger_mysql.password
+}
+
+output "ranger_rds_endpoint"{  
+  value = aws_db_instance.ranger_mysql.endpoint
+}
+
+output "ranger_rds_address"{  
+  value = aws_db_instance.ranger_mysql.address
+}
+
+#------Ranger and Solr Instance Security Groups-----
+
+#Configure Security Group for EC2
+resource "aws_security_group" "ec2_sg" {
+  name        = var.ranger_solr_sg_name
+  description = var.ranger_solr_sg_desc
+  vpc_id      = var.vpc_id
+
+  tags = {
+    Name = var.ranger_solr_sg_name
+  }
+
+  #Allow ranger admin
+  ingress {
+    from_port   = var.ranger_port
+    to_port     = var.ranger_port
+    protocol    = "tcp"
+    cidr_blocks = var.local_ips
+  }
+
+  ingress {
+    from_port       = var.ranger_port
+    to_port         = var.ranger_port
+    protocol        = "tcp"
+    cidr_blocks = [data.aws_vpc.selected.cidr_block]
+  }
+  #allow solr
+  ingress {
+    from_port       = var.solr_port
+    to_port         = var.solr_port
+    protocol        = "tcp"
+    cidr_blocks = var.local_ips
+  }
+
+  ingress {
+    from_port   = var.solr_port
+    to_port     = var.solr_port
+    protocol    = "tcp"
+    cidr_blocks = [data.aws_vpc.selected.cidr_block]
+  }
+  # SSH 
+  ingress {
+    from_port   = var.ssh_port
+    to_port     = var.ssh_port
+    protocol    = "tcp"
+    cidr_blocks = var.local_ips
+  }
+  ingress {
+    from_port   = var.ssh_port
+    to_port     = var.ssh_port
+    protocol    = "tcp"
+    cidr_blocks = [data.aws_vpc.selected.cidr_block]
+  }
+  # Allow all outbound traffic.
+  egress {
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+}
+
+##### Solr
+
+#Configure EC2 instance for solr AMI
+resource "aws_instance" "tf_solr_base_inst" {
+  ami = var.ami
+  instance_type = var.solr_instance_type
+  associate_public_ip_address = "true"
+  key_name = var.key_name
+  subnet_id = var.public_subnets[0]
+  security_groups = [aws_security_group.ec2_sg.id]
+  tags = { 
+    Name = var.ranger_solr_ami_name
+  }
+  user_data = data.template_file.solr_base_ami_tmpl.rendered
+
+  provisioner "remote-exec" {
+    connection {
+      type        = "ssh"
+      user        = "ec2-user"
+      host        = aws_instance.tf_solr_base_inst.public_dns
+      private_key = file("${path.module}/pskey.pem")
```

Comment:
pskey.pem -> is this pritishsahookey.pem :) maybe we can make this more explicit that this needs to be configured